geopandas/geodataframe.py

```diff
@@ -755,16 +765,43 @@ def concat(L, axis=0, ignore_index=False):
     geometry = vectorized.concat(df._geometry_array for df in L)
     if isinstance(L[0], GeoDataFrame):
         L = [df.drop(name, axis=1) for df in L]
-        new = pd.concat(L, ignore_index=ignore_index)
+        new = pd.concat(L, **kwargs)
         new = new.set_geometry(GeoSeries(geometry, name=name, index=new.index),
                                crs=L[0].crs)
     else:
         index = pd.concat([pd.Series(index=s.index) for s in L],
-                          ignore_index=ignore_index).index
+                          **kwargs).index
         new = GeoSeries(geometry, index=index, name=name, crs=L[0].crs)
     return new
 
 
+def _concat_axis1(objs, **kwargs):
+    is_geo = [isinstance(obj, (GeoDataFrame, GeoSeries)) for obj in objs]
+
+    if not sum(is_geo) == 1:
+        raise ValueError("'concat' with axis=1 currently only supports "
+                         "concatenating objects of which only one is a "
+                         "GeoDataFrame or Series")
```

Comment:
Why do we have this limitation?  Is it instead that we only accept one GeoDataFrame or **Geo**Series?

Reply:
Yep, the Series -> GeoSeries is a typo. 

But the actual question on the limitation: I think I mainly did this for simplicity of the implementation. Because I replace the Geo.. ones with dummy objects, and after concatting replace them. So if we have multiple of those objects, I just need to keep track of multiple and loop over them. Which should also not be that hard.